src/firebase/firestore/user.ts

```diff
@@ -9,58 +9,81 @@ import {
 } from 'firebase/firestore';
 import { UserCredential } from 'firebase/auth';
 import { db } from '../config';
-import { User } from '../../types/types';
+import { RegularUser, Admin, Employer, genericUser } from '../../types/types';
 
-const parseUser = async (document: QueryDocumentSnapshot<DocumentData>) => {
+interface Dictionary<T> {
+  [key: genericUser]: T;
+}
+
+// QueryDocumentSnapshot<DocumentData>
+const parseUser = async (document: QueryDocumentSnapshot<any>) => {
   const userId = document.id.toString();
   const data = document.data();
+  const type = data.access;
+  
+  const userConstructors: Dictionary<typeof genericUser> = {
+    regular: RegularUser,
+    admin: Admin,
+    employer: Employer,
+  }
+
+  const userType = userConstructors[type];
+
   const user = {
-    id: userId,
-    access: data.access,
-    createdJobs: data.createdJobs, // might need to map to job objects later
-    email: data.email,
-    likedJobs: data.likedJobs, // might need to map to job objects later
-    name: data.name,
-    phoneNumber: data.phoneNumber,
-    verified: data.verified,
-    password: data.password,
-  };
-  return user as User;
+    userId,
+    ...data,
+  }
+
+  return user as typeof userType;
 };
 
-export const getUser = async (id: string): Promise<User | null> => {
-  const docRef = doc(db, 'users', id);
-  const docSnap = await getDoc(docRef);
-  if (docSnap.exists()) {
-    console.log('User data:', docSnap.data());
-    return parseUser(docSnap);
-  }
-  // doc.data() will be undefined in this case
-  console.log('No such document!');
-  return null;
+export const getUser = async (id: string): Promise<genericUser | null> => {
+  const userRefs = [
```

Comment:
i would say you can extract this from the function and keep it as a constant at the top of the file